cpp/src/distance/fused_distance_nn.cu

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2021-2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cstdint>            // int64_t
+#include <raft/core/kvp.hpp>  // raft::KeyValuePair
+#include <raft/distance/fused_distance_nn-inl.cuh>
+
+#define instantiate_raft_distance_fusedDistanceNNMinReduce(DataT, OutT, IdxT) \
+  template void raft::distance::fusedDistanceNNMinReduce<DataT, OutT, IdxT>(  \
+    OutT * min,                                                               \
+    const DataT* x,                                                           \
+    const DataT* y,                                                           \
+    const DataT* xn,                                                          \
+    const DataT* yn,                                                          \
+    IdxT m,                                                                   \
+    IdxT n,                                                                   \
+    IdxT k,                                                                   \
+    void* workspace,                                                          \
+    bool sqrt,                                                                \
+    bool initOutBuffer,                                                       \
+    bool isRowMajor,                                                          \
+    raft::distance::DistanceType metric,                                      \
+    float metric_arg,                                                         \
+    cudaStream_t stream)
+
+instantiate_raft_distance_fusedDistanceNNMinReduce(double, double, int);
+instantiate_raft_distance_fusedDistanceNNMinReduce(double, double, int64_t);
```

Comment:
I am not sure that we need the instantiations for `double` type. The ANN methods are only instantiated in `float`, therefore I would recommend to only add `float` to `libraft`. 

Since we test in `double`, it does make sense to instantiate `double` for the tests. That can be done without including these in `libraft`. A similar thing is done for CAGRA: libraft only instantiate `uint32_t` index type, and we have extra `uint64_t` [instantiations](https://github.com/rapidsai/raft/blob/branch-24.04/cpp/test/neighbors/ann_cagra/search_kernel_uint64_t.cuh) for [the test](https://github.com/rapidsai/raft/blob/branch-24.04/cpp/test/neighbors/ann_cagra/test_float_int64_t.cu).